### ISSUE TITLE
Missed call notifications do not arrive

### ISSUE THREAD

**Issue by qw123wh** [SimpleMobileTools/Simple-Dialer#5]
The notification of the missed call and the call from the blocked number do not arrive.

**Comment by tibbi** [comment]
that is the responsibility of the system. And I dont want to see notifications from blocked numbers, thats why they are blocked afterall.

**Comment by QuickJack** [comment]
I have seen a similar problem. After installing Simple Dialer, the app asked me if I wanted to switch my default phone (Google Phone) to Simple Dialer. After manually adjusting some rights, the apped worked and received calls. But missed calls are still reported via Google Phone.

**Comment by qw123wh** [comment]
exact quickjack is the same thing that happens to me.  It is as if simple dialer leans on the system dialer, but if I disable the system dialer or in extreme case I uninstall it with root, goodbye missed calls I will never know if I received a missed call.

**Comment by tibbi** [comment]
you mean missed while your device was off? Or ones that you see ringing but not do anything? The notification after the ringing ends disappear?

**Comment by qw123wh** [comment]
example i mean i'm in a room and i can't hear the phone ringing in another room either because i put the phone in silence or because i'm watching tv with a very high volume etc.
Now if I have set simple dialer as the default app and have disabled or uninstalled (with root) the pre-installed dialer  app on my phone.  When I check the phone I don't see any notification telling me that I have received a missed call.

**Comment by tibbi** [comment]
aha, well that wont be added anytime soon, it needs a whole custom connectionservice implementation. The current version indeed just relies on the stock app in some way.

**Comment by QuickJack** [comment]
@tibbi: Yes, it's the notification that is received after a missed incoming call.

I think, this is an important feature. It would be great to be able to fully replace Google Phone.

**Comment by Saijin-Naib** [comment]
Reporting same behavior here with system Dialer disabled (without root, as it is allowed on my ROM)

**Comment by ltGuillaume** [comment]
> aha, well that wont be added anytime soon, it needs a whole custom connectionservice implementation. The current version indeed just relies on the stock app in some way.

Ouch, I understand that that'll take quite a bit of work, but I suppose there are lots of people who want this app as a complete replacement of the stock Dialer (for example because the stock Dialer will indefinitely bitch about Google Play Services's permissions every single time you call or get called). Those with root access will probably want only one Dialer app installed.

I hope you'll maybe reconsider the priority status of this issue because of this.

**Comment by ltGuillaume** [comment]
Any news on this? This is deal breaker for many, I'm betting.

**Comment by tibbi** [comment]
not really, it seems to work just fine to most people including me. I dont hear others complaining about it, it must be some device specific glitch.

**Comment by ltGuillaume** [comment]
Well, it's not a real replacement if it still needs the stock Phone app. And I'd say it's rather messy and RAM-unfriendly to have two apps handle your calls simultaneously.

**Comment by qw123wh** [comment]
the missed call notification arrives on the stock dialer. So today the app is not independent. The notification should arrive on your app, because if I deactivate the stock dialer = goodbye notifications of missed calls

**Comment by qw123wh** [comment]
in this app https://github.com/Chooloo/call_manage notifications work perfectly and also the problem #9  has been solved get help from this app

**Comment by Saijin-Naib** [comment]
> not really, it seems to work just fine to most people including me. I dont hear others complaining about it, it must be some device specific glitch.

I'd be surprised if it was device-specific as I doubt many have the TeraCube One like I do.

**Comment by ltGuillaume** [comment]
> > not really, it seems to work just fine to most people including me. I dont hear others complaining about it, it must be some device specific glitch.
> 
> I'd be surprised if it was device-specific as I doubt many have the TeraCube One like I do.

I'm inclined to say you're right about this. I can confirm the issue with a Samsung S5 and a Nokia 3, both running Android 9 (LineageOS and stock, resp.).

I also appreciate the fact that it will cost a lot of extra work, don't get me wrong, but I do feel this is an inherent part of a proper dialer app.

**Comment by Saijin-Naib** [comment]
Android 9 is what both Teracube I can reproduce it on have... Maybe that is the commonality?

**Comment by qw123wh** [comment]
@Saijin-Naib no it's on everyone so miui android 9, miui android 10. Aosp android 9, aosp android 10. Huawei android 7, samsung with android 8. The android version has nothing to do with it.  Not a problem with your phones

**Comment by ltGuillaume** [comment]
That's what I suspect as well.

**Comment by offa** [comment]
> Well, it's not a real replacement if it still needs the stock Phone app. And I'd say it's rather messy and RAM-unfriendly to have two apps handle your calls simultaneously.

Agreed, the current situation is really annoying. I'd love replace the stock dialer eventually, but without getting call notifications it's still not an option for me.

**Comment by meximan** [comment]
I agree. This is a "top 3 priority": otherwise there's no point in using this app if it cannot completely replace the native dialer. Otherwise, it gets super confusing.

**Comment by meximan** [comment]
I agree not knowing who you missed a call from is very important. This is a "top 3 priority": otherwise there's no point in using this app if it cannot completely replace the native dialer in rooted or custom ROM phones.

**Comment by meximan** [comment]
A good alternative in the mean time is Open Contacts which allows for encrypted contacts and records missed calls and outgoing calls. (just make sure you don't remove com.samsung.android.providers.contacts if you have a Samsung phone otherwise it'll keep crashing).

**Comment by QuickJack** [comment]
Incoming calls are handled by com.android.server.telecom(https://gossipfunda.com/com-android-server-telecom/). This works independently of the default dialer com.google.android.dialer which can be disabled. Simple Dialer logs missed incoming calls as expected. But it doesn't show any notification.

Missed incoming calls are shown on the lock screen via com.android.server.telecom. After unlocking and clicking on the notification it launches the default dialer com.google.android.dialer. It does so even if the system's default dialer entry is changed to Simple Dialer. This is the actual problem. The question is how can we manage to force com.android.server.telecom to launch Simple Dialer?

**Comment by Exilion666** [comment]
Waiting for this feature as well. At the moment stuck up on default dialer on GrapheneOS

**Comment by tibbi** [comment]
well, it just proves that Im right. Missed call notifications come from the system, not Dialer

**Comment by qw123wh** [comment]
Sorry so how do you explain that only your dialer in the whole play store has this problem? 
The dialer certainly has something to do with it, because the calls part is managed by a simple dialer, but the notifications of missed calls are managed by the system dialer, but this problem does not happen with other third-party dialers.

**Comment by tibbi** [comment]
I dont know how exactly it works, contact and call handling on Android is a huge mess that can differ a lot depending on the manufacturer. This is the latest app, of course it can have glitches.

**Comment by QuickJack** [comment]
> Maybe this can help.
> 
> https://source.android.com/devices/tech/connect/call-notification?hl=en
> 
> Please tibbi read this. Perhaps this will solve this problem

@tibbi  is right in saying that incoming call notifications are "handled by the system" ("Done by Telecom" according to the aforementioned article). But this is true for Android versions prior to 7 which corresponds to API level <24. It's important to note that Simple Dialer currently has minSdkVersion set to 23 in app/build.gradle (Android 6.x).

A lot of people are using newer smartphones and Android versions which leads to inconsistencies as described in the article. I think, it would be best to increase minSdkVersion to 24.

The provided AOSP Dialer sample app shows how all the described functionalities can be handled inside the app itself.

**Comment by meximan** [comment]
Do you take donations to prioritize certain features?

**Comment by meximan** [comment]
I would suggest using Open contacts because it shows missed calls.

**Comment by tibbi** [comment]
Im using the app with my Android 10 and it works just fine

**Comment by meximan** [comment]
I don't get a missed call notification in Android 11 from simple dialer nor simple contacts.

**Comment by ltGuillaume** [comment]
@meximan I don't know why this should be recommended: a contacts app should stay away from handling (missed) calls imho.

**Comment by meximan** [comment]
Functionality is a priority. It's OK though I am using open contacts from f-droid for the missed call functionality.

**Comment by ltGuillaume** [comment]
I meant that your (constant) recommendation of Open Contacts doesn't help.

**Comment by meximan** [comment]
It does what I need it to, so it's a solution for me. No problem I won't mention it again.

**Comment by hoi2828** [comment]
I had the same problem on my samsung phone. Trying various settings for the system phone, I found a solution that seems zu work: Set simple dialer as default app, but keep permitting system dialer access to calling lists and contacts. It works, but of course the system app can still spy my contacts, so it would be great to replace it completely by simple dialer.

**Comment by tibbi** [comment]
Simple Dialer still relies on the system, doing a fully standalone solution to everything calling related would be a nightmare. Especially as I know that such things vary a lot depending on the system version and the device manufacturer. So dont expect having any drastic changes there anytime soon.

**Comment by hoi2828** [comment]
Ok, I understand the problem. Now I also tried meximan's solution with Open Contacts installed parallelly for the missed calls functions. That also works. Ist that better for privacy than the system dialer? @meximan what do you think? I know that this question is not directly about simple dialer, but it's an issue in some cases for people who desparately want to use it ;-)

**Comment by genodeftest** [comment]
@hoi2828 : If you want to protect yourself from the default call application spying on you and sending your data to Google, you can try using [NetGuard](https://www.netguard.me/) or some other Firewall to prevent it from talking to any service on the internet.

**Comment by cicciobellohalabua2** [comment]
try to take a look at this commit:
https://github.com/android-dre/platform-packages-apps-Dialer/commit/cc42b5953588110cc89b43e72d2b6579757017a1

It is explained very superficially here too. What interests us is
“Behavior in Android 7.0 and later”

https://source.android.com/devices/tech/connect/call-notification

**Comment by biopsin** [comment]
On LinageOS-18.1 I uninstall the system dialer for user 0 and set Simple Dialer as default and that works without issues. This is governed by qualcomm in my case, so disabling com.android.phone will break the whole system services chain and that includes the system messaging service too.

**Comment by biopsin** [comment]
@cicciobellohalabua2 thats what I wanted to get across is that the missed notifications are handle by the system services not the dialer app and therefore its out of scope for SimpleMobileTools Dialer :)
